src/QueryPipeline/BlockIO.h

```diff
@@ -31,13 +31,13 @@ struct BlockIO
     /// When it is true, don't bother sending any non-empty blocks to the out stream
     bool null_format = false;
 
-    /// Call these functions if you want to log the request.
     void onFinish()
     {
         if (finish_callback)
         {
             finish_callback(pipeline);
         }
+        pipeline.reset();
```

Comment:
After adding this line, an execution of query will hang until destruction is finished. That cause a query costs apperently more time from user's perspective.

Reply:
Is it necessary to add this line? Can I remove it in my customization?

Reply:
@Alex-Cheng Is it easy to reproduce? If yes, please, provide an example. If not, give us a bit more details (what type of query and what exception, what is the additional time)

Reply:
sure, I will provide an example.

Reply:
Finally I realized it is caused by my own modification. Thank you for your time.